third_party/nvfuser/csrc/compute_at_map.cpp

```diff
@@ -1088,12 +1092,29 @@ void ComputeAtMap::buildConcreteIds() {
   // same size, it doesn't matter which is selected. This should be run-to-run
   // deterministic but which ID gets selected her depends on the traversal order
   // generating the set (compute at map build).
+  for (auto expr : ir_utils::getScatterOps(fusion_)) {
+    auto output_ids = ir_utils::allIDsOf(expr->output(0)->as<TensorView>());
+    auto index_ids = ir_utils::allIDsOf(expr->indexTv());
+    auto src_ids = ir_utils::allIDsOf(expr->srcTv());
+
+    for (size_t i = 0; i < output_ids.size(); ++i) {
+      auto out_id = output_ids[i];
+
+      auto idx_id = index_ids[i];
+      auto src_id = src_ids[i];
+      concrete_id_map_[out_id] = idx_id;
+      concrete_id_map_[src_id] = idx_id;
+    }
+  }
   for (const auto& disjoint_set_shared_ptr :
        id_graph_.exactNodes().disjointSets()) {
     TORCH_INTERNAL_ASSERT(
         disjoint_set_shared_ptr->vector().size(),
         "Cannot compute concrete id of empty set.");
     auto first_id = disjoint_set_shared_ptr->vector().front();
+    if (concrete_id_map_.find(first_id) != concrete_id_map_.end()) {
```

Comment:
Is this really necessary? I don't know why the concrete ID of the exact map needs to be changed.

Commented out and ran the C++ test. Seems to working fine.

Reply:
If we don't change concrete ID of exact map, the index of `output_tv` will be wrong. We want to generate index like this `T_output[threadIdx.x % T_index.size(0) + ...]]`. Notice that we use stride `T_index.size(0)` to access `T_output`. I have add more test cases in new commit.